DotNet UWP Samples/Earthquake Simulator/Earthquake Simulator/App.xaml.cs

```diff
@@ -0,0 +1,309 @@
+            AppServiceTriggerDetails appService = taskInstance.TriggerDetails as AppServiceTriggerDetails;
+            _appServiceDeferral = taskInstance.GetDeferral();
+            taskInstance.Canceled += OnAppServicesCanceled;
+            DataStreamerConnection = appService.AppServiceConnection;
+            DataStreamerConnection.RequestReceived += OnAppServiceRequestReceived;
+            DataStreamerConnection.ServiceClosed += AppServiceConnection_ServiceClosed;
+        }
+
+        private void OnAppServicesCanceled(IBackgroundTaskInstance sender, BackgroundTaskCancellationReason reason)
+        {
+            _appServiceDeferral.Complete();
+        }
+
+        /// <summary>
+        /// Connection with DataStreamer has been severed
+        /// </summary>
+        private async void AppServiceConnection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            await _service.Disconnect();
+
+            try
+            { 
+                await this.ViewModel.SetDataStreamerStatus("Not Connected");
+                await this.ViewModel.SetDataStreamerAppVersion("");
+                await this.ViewModel.SetDataStreamerApiVersion("");
+            }
+            catch(Exception ex)
+            {
+            }
+
+            _appServiceDeferral.Complete();
+       }
+
+        ///
+        private async void OnAppServiceRequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            // Get a deferral because we use an awaitable API below to respond to the message
+            // and we don't want this call to get cancelled while we are waiting.
+            var messageDeferral = args.GetDeferral();
+            ValueSet message = args.Request.Message;
+            ValueSet returnData = new ValueSet();
+
+            string command = message["Command"] as string;
+
+            try
+            {
+                switch (command)
+                {
+                    case "Connect":
+                        var result = await _service.Connect();
+
+                        DataStreamerConnection = sender;
+                        returnData.Add("Result", result); 
+                        await args.Request.SendResponseAsync(returnData);
+                        IsConnectedToDataStreamerAppService = true;
+
+                        try
+                        { 
+                            await this.ViewModel.SetOutputLines("");
+                            await this.ViewModel.SetDataStreamerStatus("NotReady");
+                            await this.ViewModel.SetDataStreamerApiVersion( message.ValueOrDefault("Version", "0.0").ToString());
+                            await this.ViewModel.SetDataStreamerAppVersion( message.ValueOrDefault("AppVersion", "0.0").ToString());
+                        }
+                        catch(Exception ex)
+                        {
+                        }
```

Comment:
Review your try, catch blocks to understand where we really need it vs. where not

Reply:
done